module/os/linux/zfs/zfs_ioctl_os.c

```diff
@@ -136,22 +136,18 @@ zfsdev_state_init(struct file *filp)
 	return (0);
 }
 
-static int
-zfsdev_state_destroy(struct file *filp)
+static void
+zfsdev_state_destroy(zfsdev_state_t *zs)
```

Comment:
The change of argument made it asymmetric to zfsdev_state_init().

Reply:
I'll put that back. I have started on a follow-up PR that moves zfsdev_state_init and zfsdev_state_destroy into platform-independent code since they are now nearly identical across the platforms, and in that the argument to be passed goes back to being the filep on Linux anyway.